lib/ansible/modules/cloud/digital_ocean/digital_ocean_volume_facts.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Ansible Project
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: digital_ocean_volume_facts
+short_description: Gather facts about DigitalOcean volumes
+description:
+    - This module can be used to gather facts about DigitalOcean provided volumes.
+author: "Abhijeet Kasurde (@akasurde)"
+version_added: "2.6"
+options:
+  region_name:
+    description:
+     - Name of region to restrict results to volumes available in a specific region.
+     - Please use M(digital_ocean_region_facts) for getting valid values related regions.
+    required: false
+  oauth_token:
+    description:
+     - DigitalOcean OAuth token.
+    required: true
+    aliases: ['DO_API_TOKEN', 'DO_API_KEY', 'DO_OAUTH_TOKEN']
+
+notes:
+  - Two environment variables can be used, DO_API_KEY, DO_OAUTH_TOKEN and DO_API_TOKEN.
```

Comment:
Do we want to reduce the number of environment variables? Would be nice to get to one common variable and parameter for the oauth token.

Reply:
I am no issue but it will break for those people who are using different set of environment variables.

Reply:
Yea I guess new modules we should stick with one variable, thoughts?

Reply:
I don't see a reason to keep all, they were kept in the past since new modules were created without changing previous ones.Since there are a few modules being updated / refactored and because this is part of the module_utils now, maybe we can change this *behaviour* in the next Ansbile release to not break things in minor releases.